src/services/rns/hooks/domain.hooks.ts

```diff
@@ -33,74 +34,49 @@ export default {
-                      ROW_NUMBER() OVER(
-                          PARTITION BY "tokenId"
-                          ORDER BY
-                              "creationDate" DESC
-                      ) AS "ROW_PRIORITY"
-                  FROM
-                      "rns_domain-offer"
-                  WHERE
-                  ${isOwned ? '"status" IN (\'CANCELED\', \'SOLD\')' : '"status" = \'ACTIVE\''}
-              ) "INACTIVE_OFFERS" ON "INACTIVE_OFFERS"."tokenId" = "Domain"."tokenId"
-              AND "INACTIVE_OFFERS"."creationDate" = "offers"."creationDate"
-              AND "INACTIVE_OFFERS"."ROW_PRIORITY" = 1
-          WHERE
-              "Domain"."ownerAddress" = '${ownerAddress}'
-              ${isOwned
-              ? `
-              AND ("offers"."status" IN ('CANCELED', 'SOLD') OR "offers"."tokenId" IS NULL) 
-              AND "active_offers"."tokenId" IS NULL
-              AND ("INACTIVE_OFFERS"."tokenId" IS NOT NULL OR "offers"."tokenId" IS NULL)
-              `
-              : `
-              AND "offers"."status" = 'ACTIVE'
-              `}
-              ${nameFilter?.$like ? `AND ("Domain"."name" LIKE '%${nameFilter.$like}%' OR "Domain"."tokenId" = '${numberToHex((sha3(nameFilter.$like)) as string)}')` : ''}
-          `
-
-          const sequelize = context.app.get('sequelize')
-          const results = (await sequelize.query(sql))[0]
+      (context: HookContext) => {
+        if (context.params.query) {
+          const { ownerAddress, placed, name } = context.params.query
 
-          context.result = results.map((item: Domain & DomainOffer) => {
-            const { offerId, paymentToken, price, creationDate, ...rest } = item
+          context.params.sequelize = {
+            raw: false,
+            nest: true,
+            include: [
+              {
+                model: DomainExpiration,
+                attributes: ['expirationDate']
+              },
+              {
+                model: DomainOffer,
+                as: 'offers',
+                required: false
+              }
+            ],
+            where: {
+              '$offers.tokenId$': placed
+                ? { [Op.not]: null }
+                : { [Op.is]: null },
+              ownerAddress
+            }
+          }
 
-            return {
-              ...rest,
-              offer: offerId && {
-                offerId,
-                paymentToken,
-                price,
-                creationDate
+          if (name) {
+            const { $like: nameFilter } = name
+            context.params.sequelize.where = {
+              ...context.params.sequelize.where,
+              [Op.or]: {
+                name: {
+                  [Op.like]: `%${nameFilter}%`
+                },
+                tokenId: {
+                  [Op.eq]: numberToHex(((sha3(nameFilter)) as string))
+                }
               }
             }
-          })
+          }
+          delete (context.params.query as any).status
```

Comment:
I was looking around how to do this better as I don't like the cast to `any` and discovered https://hooks-common.feathersjs.com/hooks.html#discardquery

I would suggest using this.

Reply:
ah, nice found!